Don't replace reduction init axis with new axis if bound to a thread.

csarofeen opened the pull request:
Fixes https://github.com/dmlc/tvm/issues/3407

tqchen commented:
Thanks for the PR, sorry for the delayed review as many of folks  I know in the community are traveling to FCRC recently. Can you please add a regression test case to prevent it from happening again?

csarofeen commented:
Sure, I should be able to base it off the repro in the issue. I'll put it on my todo list but may not get to it for a while. I've been prototyping some kernels in TE, which is why the few small bug fixes. When I get a break I'll come back and write some tests.

tqchen commented:
@csarofeen can you add a test case?

tqchen commented:
ping @csarofeen

csarofeen commented:
Sorry this was so delayed, was actually preparing a talk on TE for CUDA code generation. Simple test added @tqchen

csarofeen commented:
I can't tell if the linter error is from my change or not. I tried to get it running locally but I got a docker build error.

merrymercy commented on tests/python/unittest/test_codegen_cuda.py:
```diff
@@ -154,9 +154,25 @@ def check_inf_nan(ctx, n, value, dtype):
+
+    mo, _ = s[B].split(B.op.axis[0], 32)
+    s[B].bind(mo, tvm.thread_axis("blockIdx.x"))
+    fcuda = tvm.build(s, [A, B], "cuda")
```
Add guard to this test (see Line 143). Because some test environments do not have a gpu.

csarofeen replied:
Thanks, updated.

csarofeen commented:
@merrymercy @tqchen I rebased at top of tree and added 2 tests to cover: https://github.com/dmlc/tvm/pull/3444 and https://github.com/dmlc/tvm/pull/3382 hope that's okay.

csarofeen commented:
Ping. If this looks good I can rebase so it can be merged.

tqchen commented:
Thanks @csarofeen please rebase

csarofeen commented:
Looks like this got stuck, is there a way to retry the tests?

tqchen commented:
Thanks @csarofeen , this PR is now merged!
